packages/client/README.md

```diff
@@ -0,0 +1,74 @@
+# `@lens-protocol/client`
+
+The official framework-agnostic Lens API Client.
+
+This package enables you to interact with the Lens API via a type safe interface that abstracts away some of the GraphQL intricacies.
+
+> **Warning**
+>
+> The Lens SDK is still in its initial development phase. Anything MAY change at any time.
+> This is a Developer Preview aimed primarily at existing integrators so to gather [early feedback](https://github.com/lens-protocol/lens-sdk/discussions/48).
+
+## Examples
+
+For all exaples below we require the same setup, we need a LensClient instance and an ethers.js Wallet as a signer.
+
+```ts
+import LensClient, { mumbai } from '@lens-protocol/client';
+import { Wallet } from 'ethers';
+
+const lensClient = LensClient.init(mumbai);
+
+const wallet = new Wallet(
+  'YOUR_PRIVATE_KEY', // keep it secret (!) ie. using env vars
+);
+
+const address = await wallet.getAddress();
+```
+
+### Authenticate with Lens API
+
+```ts
+const challenge = await lensClient.generateChallenge(address);
+const signature = await wallet.signMessage(challenge);
+
+const { accessToken, refreshToken } = await lensClient.authenticate(address, signature);
+
+// lensClient is ready for authenticated requests, you can store refreshToken for later use
+```
+
+### Verify credentials and reauthenticate from stored credentials
+
+```ts
+const storedCredentials = {
+  accessToken: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXV...',
+  refreshToken: 'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXV...',
+};
+
+const isValid = await lensClient.isAccessTokenValid(storedCredentials.accessToken);
+
+if (!isValid) {
+  const { accessToken, refreshToken } = await lensClient.refreshCredentials(
```

Comment:
I was initially advocating for this as well. One caveat to notice is that the consumer is responsible for storing these tokens, so somehow they are exposed about the details of`ICredentials` structure cause they either simply assert with it: `JSON.parse(serialized) as ICredentials` OR they use proper parsing (e.g. via zod).

I noticed a small mistake in the current "resume from stored tokens flow".

Let's say the consumer app resumes in a fresh thread from stored credentials (new tab for a SPA app, new process for a NodeJS app, hard refresh on a web app)... what happen if the `lensClient.isAccessTokenValid(accessToken)` returns true?

How does the `LensClient` can populate the `credentials` private member?

For context, the whole conversation about 2 static methods (now being just `init`) was part of a brainstorming session between me and @krzysu. The second method would have allowed to create a `LensClient` out of stored credentials.


A small snippet of code from that session to illustrate the point.
```ts

// this could be opaque to the end user
type Credentials = {
  accessToken: string
  refreshToken: string
  environment: Environment
}

// an example usage on a web app
function createLensClient(): LensClient {
  const serialized = localStorage.get('credentials') ?? null
  
  if (serialized) {
    const credentials = JSON.parse(serialized) // overly simplified to reduce noise
    return LensClient.fromCredentials(credentials)
  }

  return LensClient.init(environments.production)
}
```

The `Credentials` type was not 1:1 with the returned value from the API call, but somewhat an augmented object that was also encapsulating the environment details for which those credentials are valid. In the end they won't work in any other environment and that was a cheap solution to avoid possible mismatch between the specified environment details and stored credentials, but this is marginal for this conversation.

Now, I am not advocating to go back to this approach necessarily. I think we have good potential with current "single `init` factory method (or simply just `LensClient` constructor as it was suggested by @krzysu in a comment above).

An interesting alternative, which is reminiscent of the way we handle storing of auth tokens in the `@lens-protocol/react` package, is to just have the consumer deal with the Refresh Token.

An advantage of that is that not being a "complex structure" but just a string... it can be properly dealt with as opaque content. No need for TS assertion after recovering it from storage nor parsing, it's just an opaque string.


What does it mean for the initial authentication? Not much TBH
```ts
const token = await client.authenticate(address, signature)
```
Consumer responsibility is to just store it, they can ignore it's a JWT, they can ignore the fact it's the Refresh Token... it's all opaque to them. Easy.

On a start/restart from stored credentials (keep using the web app example above but can be adapted to other storage solutions):
```ts
const token = localStorage.get('token') ?? null

if (token) {
  const newToken = await client.refreshCredentials(token)
  
  localStorage.set(token, newToken)
}
// continue as usual
```

This approach promotes that each thread gets a fresh copy of Access Token... this will reduce the chances for the token to expire while in use and probably for a good class of applications interested in using the `LensClient` they might not even care much about it expiring as the expiry time could well be outside of their normal operating timeframe (think about the request lifecycle from a request handler on a server app).

Reply:
As Access Token is not longer exposed... the `isAccessTokenValid` would need to change into just a check on an internal, that could be used to either inform the consumer it's time to try a `refreshCredentials` and if it also fails a brand new re-authentication.

```ts
const isAuthenticated = await client.isAuthenticated()
```

Reply:
Pretty like the way to not expose token and use the same patter with `IStorageProvider` that we already make our sdk users familiar with

Reply:
after a long discussion, we settled on an approach to managing the token lifecycle and storage inside the client, ticket was created, it is now outside of the scope of this PR
https://avara.height.app/T-5607
I will leave these two methods in this PR

```
client.generateChallenge(address)
client.authenticate(address, signature) // returns nothing
```